Credit Card Validation/src/main/java/CreditCardValidator.java

```diff
@@ -1,94 +1,65 @@
-  
-  private final String paymentName;
-  private final List<String> prefix;
-  
-  CardType(String paymentName, String ...prefix) {
-    this.paymentName = paymentName;
-    this.prefix = new ArrayList<String>(List.of(prefix));
-  }
-  
-  public List<String> getPrefix() {
-    return this.prefix;
-  }
-  public String getPaymentName() {
-    return this.paymentName;
-  }
-}
-
 public class CreditCardValidator {
   private final int CREDIT_CARD_LENGTH = 16;
-  private final String creditCardNumber;
 
-  public CreditCardValidator(String creditCardNumber) {
-    this.creditCardNumber = creditCardNumber;
+  public String validateNumbers(String creditCardNumber) {
+    boolean letterFound = hasLetter(creditCardNumber);
+
+    if (!letterFound) {
+      String[] cardDigitsArray = creditCardNumber.split(" ");
+      if (cardDigitsArray.length == 4) {
+        return PaymentSystemResolver(cardDigitsArray[0]);
+      } else if (creditCardNumber.length() == 16) {
+        return PaymentSystemResolver(creditCardNumber.substring(0, 4));
+      } else {
+        return "";
+      }
+    }
+    return "";
   }
 
-  public String validateNumbers() {
+  private boolean hasLetter(String creditCardNumber) {
     for (int i = 0; i < creditCardNumber.length(); i++) {
       if (Character.isWhitespace(creditCardNumber.charAt(i)))
         continue;
 
       if (!Character.isDigit(creditCardNumber.charAt(i))) {
-        return null;
+        return true;
       }
     }
 
-    String[] cardDigitsArray = creditCardNumber.split(" ");
-    if (cardDigitsArray.length == 4) {
-      return PaymentSystemResolver(cardDigitsArray[0]);
-    } else if (creditCardNumber.length() == 16) {
-      return PaymentSystemResolver(creditCardNumber.substring(0, 4));
-    } else {
-      return null;
-    }
+    return false;
   }
 
   private String PaymentSystemResolver(String prefix) {
-    String systemName;
+    String systemName = "";
     String prefixFirstTwo = prefix.substring(0, 2);
 
-    if (CardType.VISA.getPrefix().contains(prefix.substring(0,1)))
-      systemName = CardType.VISA.getPaymentName();
-    else if (CardType.MASTERCARD.getPrefix().contains(prefixFirstTwo))
-      systemName = CardType.MASTERCARD.getPaymentName();
-    else if (CardType.JCB.getPrefix().contains(prefixFirstTwo))
-      systemName = CardType.JCB.getPaymentName();
-    else if (CardType.DISCOVER.getPrefix().contains(prefix) || CardType.DISCOVER.getPrefix().contains(prefixFirstTwo))
-      systemName = CardType.DISCOVER.getPaymentName();
-    else if (CardType.AMERICAN_EXPRESS.getPrefix().contains(prefixFirstTwo))
-      systemName = CardType.AMERICAN_EXPRESS.getPaymentName();
-    else
-      systemName = null;
+    for (CardType cardType : CardType.values()) {
+      if (cardType.getPrefixs().contains(prefix.substring(0, 1)) || cardType.getPrefixs().contains(prefixFirstTwo)) {
+        systemName = cardType.getPaymentSystemName();
+      }
+    }
 
     return systemName;
   }
 
```

Comment:
What I can recommend:
- Method "validateCreaditCard" should collect all errors and return list of errors OR throw exception which contains all errors.
- Each validation should be done separately.
- Don't mix responsibilities! validateCreditcard validate card AND write information to the screen.
- You was not carefully with previous PR. Please, if something is not clear - came to me at Skype and ask questions. Each comment must be clear for you and fix must be applied:)